src/gl/canvas2dRe.re

```diff
@@ -64,7 +68,14 @@ module LineJoin: LineJoinType = {
   let miter : t = "miter";
 };
 
+type image 'a =
+  | Number: image float
+  | ImageData: image ImageRe.t;
 
+type style _ =
```

Comment:
Varaints will be "boxed" in an array, so that unfortunately won't work properly. You'll have to use it as a phantom argument to be able to pass it directly as a value to the external. And the getter should then just return a tuple, like `reifyType`.